tw-tasks-core/src/main/java/com/transferwise/tasks/triggering/KafkaTasksExecutionTriggerer.java

```diff
@@ -369,49 +371,79 @@ void releaseCompletedOffset(ConsumerBucket consumerBucket, TopicPartition topicP
     }
   }
 
-  private void commitOffsets(ConsumerBucket consumerBucket, boolean finalize) {
+  private void commitSync(ConsumerBucket consumerBucket, Map<TopicPartition, OffsetAndMetadata> offsetsToCommit) {
+    if (offsetsToCommit.isEmpty()) {
+      return;
+    }
+
+    var bucketId = consumerBucket.getBucketId();
+    var success = false;
+
+    try {
+      if (log.isDebugEnabled()) {
+        log.debug("Sync-committing bucket '" + bucketId + "' offsets to Kafka: " + offsetsToCommit.entrySet().stream()
+            .map(e -> e.getKey() + ":" + e.getValue().offset()).collect(Collectors.joining(", ")));
+      }
+      consumerBucket.getKafkaConsumer().commitSync(offsetsToCommit);
+    } catch (Throwable t) {
+      registerCommitException(bucketId, t);
+    } finally {
+      coreMetricsTemplate.registerKafkaTasksExecutionTriggererCommit(bucketId, true, success);
```

Comment:
Where is the success boolean set to true?

Reply:
Fixed.